src/artemis/fgs_communicator.py

```diff
@@ -0,0 +1,144 @@
+import os
+import time
+
+from bluesky.callbacks import CallbackBase
+
+import artemis.log
+from artemis.ispyb.store_in_ispyb import StoreInIspyb2D, StoreInIspyb3D
+from artemis.nexus_writing.write_nexus import (
+    NexusWriter,
+    create_parameters_for_first_file,
+    create_parameters_for_second_file,
+)
+from artemis.parameters import FullParameters
+from artemis.zocalo_interaction import run_end, run_start, wait_for_result
+
+
+class FGSCommunicator(CallbackBase):
```

Comment:
Should: As discussed i think breaking these up into the different communicators would be more reusable. See comment at https://github.com/DiamondLightSource/python-artemis/issues/164#issuecomment-1262072452. It probably then makes sense for each communicator to be in the appropriate file e.g. the zocalo one in `zocalo_interaction`

Reply:
I agree it makes sense to do this, but I think we should discuss a bit more about how to organise it so I'd rather do that in a separate issue/PR

Reply:
done, #315